python/lsst/cp/pipe/ptc.py

```diff
@@ -365,10 +376,102 @@ def runDataRef(self, dataRef, visitPairs):
         dataRef.put(dataset, datasetType="photonTransferCurveDataset")
 
+        self.log.info(f"Writing linearizers")
+
+        detName = detector.getName()
+        now = datetime.datetime.utcnow()
+        butler = dataRef.getButler()
+
+        for linPair in [("LOOKUPTABLE", 'linearizeLut'), ("LINEARIZEPOLYNOMIAL", 'linearizePolynomial'),
+                        ("LINEARIZESQUARED", 'linearizeSquared')]:
+
+            linType = linPair[0]
+            dataType = linPair[1]
+            if linType == "LOOKUPTABLE":
+                tableArray = lookupTableArray
+            else:
+                tableArray = None
+
+            linearizer = self.buildLinearizerObject(dataset, detName, detNum, detector,
+                                                    instruName=self.config.instrumentName,
+                                                    linearizerType=linType, tableArray=tableArray,
+                                                    log=self.log)
+            butler.put(linearizer.toDict(), datasetType=dataType, dataId={'detector': detNum,
+                       'detectorName': detName, 'calibDate': now.strftime("%Y-%m-%d")})
+
         self.log.info('Finished measuring PTC for in detector %s' % detNum)
 
         return pipeBase.Struct(exitStatus=0)
 
+    def buildLinearizerObject(self, dataset, detName, detNum, detector, instruName='', linearizerType='',
+                              tableArray=None, filt='NONE', log=None):
+        """Build linearizer object to persist.
+
+        Parameters
+        ----------
+        dataset : `lsst.cp.pipe.ptc.PhotonTransferCurveDataset`
+            The dataset containing the means, variances and exposure times
+        detName : `srt`
+            Detector name
+        detNum : `int`
+            Detector number
+        detector : `lsst.afw.cameraGeom.detector.detector.Detector`
+            Detector object
+        instruName : `str`, optional
+            Instrument name
+        linearizerType : `str`, optional
+            'LOOKUPTABLE' or 'LINEARIZESQUARED'
+        tableArray : `np.array`, optional
+            Look-up table array with size rows=nAmps and columns=ADU values
+        filt : filter, `str`, optional
+        log : `lsst.log.Log`, optional
+            Logger to handle messages
+
+        Returns
+        -------
+        linearizer : `lsst.ip.isr.Linearizer`
+            Linearizer object
+        """
+        if linearizerType == 'LOOKUPTABLE':
+            if tableArray is not None:
+                linearizer = Linearizer(table=tableArray, log=log)
+            else:
+                raise RuntimeError("tableArray must be provided when creating a LookupTable linearizer")
+        elif linearizerType in ['LINEARIZESQUARED', 'LINEARIZEPOLYNOMIAL']:
+            linearizer = Linearizer(log=log)
+        else:
+            raise RuntimeError("Enter supporter linearizerType to build a Linearizer object: " +
```

Comment:
Typo: supporter -> supported. Also th error message should say what was provided that caused this error. Finally remove the `+` from the end since it's not needed.